Flaky test: "replication_set_should_not_change_if_ring_did_not_change,_when_num_of_instance_per_zone_is_inconsistent"

Noticed build failure: https://github.com/cortexproject/cortex/actions/runs/12798522991/job/35682746078?pr=6511

```
--- FAIL: TestRing_Get_Consistency (488.91s)
    --- FAIL: TestRing_Get_Consistency/replication_set_should_not_change_if_ring_did_not_change,_when_num_of_instance_per_zone_is_inconsistent (61.47s)
        ring_test.go:735: 
            	Error Trace:	/__w/cortex/cortex/pkg/ring/ring_test.go:735
            	Error:      	"1" is not less than or equal to "0"
            	Test:       	TestRing_Get_Consistency/replication_set_should_not_change_if_ring_did_not_change,_when_num_of_instance_per_zone_is_inconsistent
FAIL
FAIL	github.com/cortexproject/cortex/pkg/ring	704.951s
```

--- a/pkg/ring/model.go
+++ b/pkg/ring/model.go
@@ -479,7 +479,14 @@ func (d *Desc) Clone() interface{} {
 func (d *Desc) getTokensInfo() map[uint32]instanceInfo {
 	out := map[uint32]instanceInfo{}
 
-	for instanceID, instance := range d.Ingesters {
+	keys := []string{}
+	for key, _ := range d.Ingesters {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	for _, instanceID := range keys {
+		instance := d.Ingesters[instanceID]
 		info := instanceInfo{
 			InstanceID: instanceID,
 			Zone:       instance.Zone,

--- a/pkg/ring/model.go
+++ b/pkg/ring/model.go
@@ -479,13 +479,13 @@ func (d *Desc) Clone() interface{} {
 func (d *Desc) getTokensInfo() map[uint32]instanceInfo {
 	out := map[uint32]instanceInfo{}
 
-	keys := []string{}
-	for key, _ := range d.Ingesters {
-		keys = append(keys, key)
+	instanceIDs := []string{}
+	for key := range d.Ingesters {
+		instanceIDs = append(instanceIDs, key)
 	}
-	sort.Strings(keys)
+	sort.Strings(instanceIDs)
 
-	for _, instanceID := range keys {
+	for _, instanceID := range instanceIDs {
 		instance := d.Ingesters[instanceID]
 		info := instanceInfo{
 			InstanceID: instanceID,

